src/main/java/se/simonsoft/cms/indexing/properties/HandlerSubjectScheme.java

```diff
@@ -0,0 +1,151 @@
+import se.simonsoft.cms.indexing.xml.custom.ContentHandlerToIndexFields;
+import se.simonsoft.cms.indexing.xml.custom.LoggingErrorListener;
+import se.simonsoft.cms.item.CmsItemId;
+import se.simonsoft.cms.item.CmsItemPath;
+import se.simonsoft.cms.item.CmsRepository;
+import se.simonsoft.cms.item.RepoRevision;
+import se.simonsoft.cms.item.events.change.CmsChangesetItem;
+import se.simonsoft.cms.item.info.CmsItemNotFoundException;
+import se.simonsoft.cms.item.properties.CmsItemProperties;
+import se.simonsoft.cms.xmlsource.handler.XmlSourceReader;
+import se.simonsoft.cms.xmlsource.handler.s9api.XmlSourceDocumentS9api;
+import se.simonsoft.cms.xmlsource.handler.s9api.XmlSourceElementS9api;
+import se.simonsoft.cms.xmlsource.handler.s9api.XmlSourceReaderS9api;
+import se.simonsoft.cms.xmlsource.transform.SaxonMessageListener;
+
+import javax.inject.Inject;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.stream.StreamSource;
+import java.util.*;
+import java.util.regex.Pattern;
+
+public class HandlerSubjectScheme extends HandlerLogicalId {
+
+    private final XmlSourceReaderS9api sourceReader;
+    private ItemContentBufferStrategy contentStrategy;
+    private transient XsltTransformer transformer;
+
+    private static final Logger logger = LoggerFactory.getLogger(HandlerSubjectScheme.class);
+
+    @Inject
+    public HandlerSubjectScheme(XmlSourceReader sourceReader, Processor processor) {
+        var stylesheet = new StreamSource(this.getClass().getClassLoader().getResourceAsStream("se/simonsoft/cms/indexing/properties/subject-scheme-props.xsl"));
+        this.sourceReader = (XmlSourceReaderS9api) sourceReader;
+        XsltCompiler compiler = processor.newXsltCompiler();
+        XsltExecutable xsltCompiled;
+        try {
+            logger.info("Compiling subject scheme properties indexing XSL.");
+            xsltCompiled = compiler.compile(stylesheet);
+            logger.info("Compiled subject scheme properties indexing XSL.");
+        } catch (SaxonApiException e) {
+            throw new RuntimeException("Failed to compile subject scheme properties indexing XSL: " + e.getMessage(), e);
+        }
+        transformer = xsltCompiled.load();
+    }
+
+    @Inject
+    public void setItemContentBufferStrategy(ItemContentBufferStrategy contentStrategy) {
+        this.contentStrategy = contentStrategy;
+    }
+
+    @Override
+    public void handle(IndexingItemProgress progress) {
+        String excludePattern = "(cmsconfig.*:.*|svn:.*|abx:.*)";
+        CmsItemProperties itemProperties = progress.getProperties();
+        CmsItemId itemId = getItemId(progress);
+        if (itemId == null) return;
+        CmsItemPath itemRelPath = itemId.getRelPath();
+        if (itemRelPath == null) return;
+        CmsItemPath itemParentPath = itemRelPath.getParent();
+        if (itemParentPath == null) return;
+        XdmMap properties = new XdmMap();
+        for (String key : itemProperties.getKeySet()) {
```

Comment:
This is deep into Saxon APIs compared to getPropertiesMetaForm in se.simonsoft.cms.release.properties.PropertiesServiceImpl. But I think you are right that we cannot use our TransformerService with a custom SAXDestination.

Reply:
Yes using the TransformerService was my first attempt but like you said, I couldn't provide my a custom SAXDestination. TransformerService provides no transform methods that take custom Destinations.